velox/docs/functions/presto/decimal.rst

```diff
@@ -174,6 +174,39 @@ digits after the decimal point, hence, max(s1, s2).
 
     SELECT 1.2 / 0.01
 
+Modulus
+-------
+
+For the modulus operation :code:`a % b`, when a and b are integers, the result
+`r` is less than `b` and less than or equal to `a`. Hence the number of digits
+needed to represent `r` is no more than the minimum of the number of digits
+needed to represent `a` or `b`. We can extend this to decimal inputs `a` and
+`b` by computing the modulus of their unscaled values. However, we should
+first make sure that `a` and `b` have the same scale. This can be achieved by
+scaling up the input with lesser scale by the difference in the inputs' scales,
+so both `a` and `b` have scale s. Once `a` and `b` have the same scale, we
+compute the modulus of their unscaled values, A and B. `r` has s digits after
+the decimal point, and since `r` does not need any more digits than the
+minimum number of digits needed to represent `a` or `b`, the result precision
+needs to be increased by the smaller of the differences in the precision and
+scale of either inputs. Hence the result type precision and scale are:
+
+::
+
+    s = max(s1, s2)
+    p = min(p2 - s2, p1 - s1) + max(s1, s2)
+
+To compute R, we first rescale A and B using rescale factors :code:`max(s2 - s1, 0)`
+and :code:`max(s1 - s2, 0)` respectively. Then we compute the modulus of the
+unscaled values and ensure there are 's' digits after the decimal point.
+
+::
+
+    a = (A / 10^s1) * 10^(max(s2 - s1, 0)) = (A / 10^s)
```

Comment:
It is confusing to reuse A and B for original and rescaled values. Perhaps, use A and B for original values and A' and B' for rescaled values. Also, `(A / 10^s1) * 10^(max(s2 - s1, 0))` is not very readable, is it?

Consider something like this:

```
To compute R, first rescale A and B to 's':

A = a * 10^s1
B = b * 10^s2

A' = a * 10^s
B' = b * 10^s

Then, compute modulus of the rescaled values:

R = A' % B' = r * 10^s

For example, given a = 12.3 and b = 1.21, s = max(1, 2) = 2 and p = min(2, 1) + s = 3.

A = 12.3 * 10 = 123
B = 1.12 * 100 = 112

A' = 1230
B' = 112

R = 1230 % 112 = 20 = 0.20 * 100
```